exporter/opensearchexporter/factory.go

```diff
@@ -0,0 +1,58 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package opensearchexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/opensearchexporter"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "opensearch"
+	// The stability level of the exporter.
+	stability = component.StabilityLevelDevelopment
+)
+
+// NewFactory creates a factory for OpenSearch exporter.
+func NewFactory() exporter.Factory {
+	return exporter.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		exporter.WithTraces(createTracesExporter, stability),
+	)
+}
+
+func createDefaultConfig() component.Config {
+	return &Config{
+		HTTPClientSettings: HTTPClientSettings{
+			Timeout: 90 * time.Second,
+		},
+		Namespace: "",
+		Dataset:   "",
```

Comment:
These don't appear to align with the defaults documented in the README.

Reply:
Good catch. Updated and fixed in e240c46aca23e87a1420d36e615d2e5aec6e84bc.